File: atcoder.jp/C++ (GCC 9.2.1)/abc266/abc266_c.cpp
#include <iostream>
using namespace std;


bool in_triangle(int Ax, int Ay, int Bx, int By, int Cx, int Cy, int Px, int Py) {
    int c1 = (Bx - Ax) * (Py - Ay) - (By - Ay) * (Px - Ax);
    int c2 = (Cx - Bx) * (Py - By) - (Cy - By) * (Px - Bx);
    int c3 = (Ax - Cx) * (Py - Cy) - (Ay - Cy) * (Px - Cx);
    return c1 >= 0 && c2 >= 0 && c3 >= 0 || c1 <= 0 && c2 <= 0 && c3 <= 0;
}


int main()
{
    int Ax, Ay, Bx, By, Cx, Cy, Dx, Dy;
    cin >> Ax >> Ay >> Bx >> By >> Cx >> Cy >> Dx >> Dy;
    
    if (in_triangle(Ax, Ay, Bx, By, Cx, Cy, Dx, Dy) ||
        in_triangle(Ax, Ay, Bx, By, Dx, Dy, Cx, Cy) ||
        in_triangle(Cx, Cy, Dx, Dy, Ax, Ay, Bx, By) ||
        in_triangle(Cx, Cy, Dx, Dy, Bx, By, Ax, Ay)) {
        cout << "No" << endl;
    } else {
        cout << "Yes" << endl;
    }

    return 0;
}

File: atcoder.jp/Rust (1.42.0)/abc266/abc266_c.rs
use proconio::{fastout, input};

#[fastout]
fn main() {
    input! {
        (ax, ay, bx, by, cx, cy, dx, dy): (i64, i64, i64, i64, i64, i64, i64, i64),
    }

    if in_triangle(ax, ay, bx, by, cx, cy, dx, dy) ||
        in_triangle(ax, ay, bx, by, dx, dy, cx, cy) || 
        in_triangle(ax, ay, dx, dy, cx, cy, bx, by) ||
        in_triangle(dx, dy, bx, by, cx, cy, ax, ay)
    {
        println!("No");
    }
    else
    {
        println!("Yes");
    }
    
}   


fn in_triangle(ax: i64, ay: i64, bx: i64, by: i64, cx: i64, cy: i64, px: i64, py: i64) -> bool {
    let c1: i64 = (bx - ax) * (py - ay) - (by - ay) * (px - ax);
    let c2: i64 = (cx - bx) * (py - by) - (cy - by) * (px - bx);
    let c3: i64 = (ax - cx) * (py - cy) - (ay - cy) * (px - cx);
    return c1 >= 0 && c2 >= 0 && c3 >= 0 || c1 < 0 && c2 < 0 && c3 < 0;
}